source/dem/particle_particle_broad_search.cc

```diff
@@ -201,16 +203,19 @@ ParticleParticleBroadSearch<dim>::find_particle_particle_contact_pairs(
         {
           // Get mobility status of current neighbor cell
           unsigned int neighbor_cell_mobility_status =
-            disable_contacts_object.check_cell_mobility(
-              *cell_neighbor_iterator);
-
-          // No storing of particles as candidate if main cell is active,
-          // but neighbor is not mobile (active or inactive).
-          // Particle contacts between particles in 2 active cells or with
-          // particles in inactive cells are irrelevant.
-          // In this case, we skip this neighbor cell.
-          if (main_cell_mobility_status == DisableContacts<dim>::active &&
-              neighbor_cell_mobility_status != DisableContacts<dim>::mobile)
+            sparse_contacts_object.check_cell_mobility(*cell_neighbor_iterator);
+
+          // No storing of particles as candidate if main cell has any active
+          // status, but neighbor is not mobile (active or inactive). Particle
+          // contacts between particles in 2 active cells or with particles in
+          // inactive cells are irrelevant. In this case, we skip this neighbor
+          // cell.
+          if ((main_cell_mobility_status ==
```

Comment:
Since you have some checks that match, could you do it as a "cascade"? For instance, the first check at lines 213 and 274 match. This way you would prevent rechecking, reducing the size of this part, and improving readability.

Reply:
I don't understand, it doesn't sound like what we talk yesterday. I get the cascade part, but not the match at 213 and 274 since it in 2 different loops.